src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -0,0 +1,205 @@
+    (validatedOptions |@| backendOptions) { (options, b) => options }
+  }
+
+  private def validateRawInputs(id: WorkflowId, json: WdlJson): ErrorOr[Map[String, JsValue]] = {
+    Try(json.parseJson) match {
+      case Success(JsObject(inputs)) => inputs.successNel
+      case _ => s"Workflow ${id.toString} contains bad inputs JSON: $json".failureNel
+    }
+  }
+
+  private def validateCoercedInputs(id: WorkflowId,
+                                    rawInputs: Map[String, JsValue],
+                                    namespace: NamespaceWithWorkflow): ErrorOr[WorkflowCoercedInputs] = {
+    namespace.coerceRawInputs(rawInputs) match {
+      case Success(r) => r.successNel
+      case Failure(e) => e.getLocalizedMessage.failureNel
+    }
+  }
+
+  private def validateBackendOptions(id: WorkflowId, options: WorkflowOptions, backend: Backend): ErrorOr[Unit] = {
+    try {
+      backend.assertWorkflowOptions(options)
+      ().successNel
+    } catch {
+      case e: Exception =>
+        s"Workflow ${id.toString} has invalid options for backend ${backend.backendType}: ${e.getLocalizedMessage}".failureNel
+    }
+  }
+
+  private def validateDeclarations(id: WorkflowId,
+                                   namespace: NamespaceWithWorkflow,
+                                   options: WorkflowOptions,
+                                   coercedInputs: WorkflowCoercedInputs,
+                                   backend: Backend): ErrorOr[WorkflowCoercedInputs] = {
+    val ioInterface = backend.ioInterface(options)
+    namespace.staticDeclarationsRecursive(coercedInputs, backend.engineFunctions(ioInterface)) match {
+      case Success(d) => d.successNel
+      case Failure(e) => s"Workflow ${id.toString} has invalid declarations: ${e.getLocalizedMessage}".failureNel
+    }
+  }
+
+  private val DefaultCallCachingValue = false
+  private val DefaultLookupDockerHash = false
+
+  private def disabledMessage(readWrite: String, consequence: String) =
+    s"""$readWrite is enabled in the workflow options but Call Caching is disabled in this Cromwell instance.
+       |As a result the calls in this workflow $consequence
+       """.stripMargin
+
+  val writeDisabled = disabledMessage("Write to Cache", "WILL NOT be cached")
+  val readDisabled = disabledMessage("Read from Cache", "WILL ALL be executed")
+
+  // TODO: Add to lenthall
+  def getConfigOption(conf: Config, key: String): Option[Config] = if (conf.hasPath(key)) Option(conf.getConfig(key)) else None
+
+  def configCallCaching(conf: Config) = lookupWithDefault(conf, "call-caching", "enabled", DefaultCallCachingValue)
+  def lookupDockerHash(conf: Config) = lookupWithDefault(conf, "call-caching", "lookup-docker-hash", DefaultLookupDockerHash)
+  def lookupWithDefault(conf: Config, stanza: String, key: String, default: Boolean) = {
+    (for {
+      config <- getConfigOption(conf, stanza)
+      value <- config.getBooleanOption(key)
+    } yield value) getOrElse default
+  }
```

Comment:
I added some comments as when I started looking at doing what I was talking about doing, there were multiple cans of worms I saw opening up. Since this change itself was a can of worms which came out of a can of worms i decided to avoid future cans of worms for now.
![5124hyepjal](https://cloud.githubusercontent.com/assets/961771/11913841/9a6e2de8-a63f-11e5-82df-256113bc34e6.jpg)